editor/src/core/workers/parser-printer/parser-printer.spec.ts

```diff
@@ -4822,61 +4827,109 @@ export var whatever2 = (props) => <View data-uid='aaa'>
+              }
             }
-          }
-          const elementUIDS = new Set(uids)
-          const highlightBoundsUIDS = new Set(Object.keys(success.highlightBounds))
-          return setsEqual(elementUIDS, highlightBoundsUIDS)
-        },
-        (unparsed) => {
-          console.error(unparsed)
-          return false
-        },
-        parseResult,
+
+            // Check the UIDs for the elements, which excludes attributes.
+            const elementUIDS = new Set(uids)
+            const highlightBoundsUIDs = new Set(Object.keys(success.highlightBounds))
+            if (!setsEqual(elementUIDS, highlightBoundsUIDs)) {
+              throw new Error(
+                `Element UIDs [${Array.from(
+                  elementUIDS,
+                ).sort()}] do not match the highlight bounds UIDs: [${Array.from(
+                  highlightBoundsUIDs,
+                ).sort()}]`,
+              )
+            }
+
+            // Check that this parse has not surfaced any duplicates within itself.
+            const fileValue = textFile(
+              textFileContents(printedCode, success, RevisionsState.ParsedAhead),
+              null,
+              null,
+              1234,
+            )
+            const uniqueIDsResult = getAllUniqueUids(contentsToTree({ '/index.js': fileValue }))
+            const anyDuplicates = Object.keys(uniqueIDsResult.duplicateIDs).length > 0
+            if (anyDuplicates) {
+              throw new Error(`Found duplicate UIDs: ${uniqueIDsResult.duplicateIDs}`)
+            }
+
+            // Check the UIDs for anything and everything, including attributes.
+            const fullHighlightBoundsUIDs = new Set(Object.keys(success.fullHighlightBounds))
+            const allUIDsAreEqual = setsEqual(
+              fullHighlightBoundsUIDs,
+              new Set(uniqueIDsResult.uniqueIDs),
+            )
+            if (!allUIDsAreEqual) {
+              throw new Error(
+                `All UIDs [${Array.from(
+                  uniqueIDsResult.uniqueIDs,
+                ).sort()}] do not match the full highlight bounds UIDs: [${Array.from(
+                  fullHighlightBoundsUIDs,
+                ).sort()}]`,
+              )
+            }
+          },
+          (unparsed) => {
+            throw new Error(`${unparsed}`)
+          },
+          parseResult,
+        )
+        return true
+      }
+      const printableArbitrary = printableProjectContentArbitrary()
+      const dataUIDProperty = FastCheck.property(
+        printableArbitrary,
+        checkElementUIDSMatchHighlightBounds,
       )
+      FastCheck.assert(dataUIDProperty, { verbose: false, numRuns: 100 })
     }
-    const printableArbitrary = printableProjectContentArbitrary()
-    const dataUIDProperty = FastCheck.property(
-      printableArbitrary,
-      checkElementUIDSMatchHighlightBounds,
-    )
-    FastCheck.assert(dataUIDProperty, { verbose: true })
+    /* eslint-disable-next-line jest/expect-expect */
```

Comment:
Rather than do this we can provide the names of the functions that will include the `expect` like so https://github.com/concrete-utopia/utopia/blob/f6ab2ccf0ea6bc78e4c82988f8f86be95d80b170/editor/src/components/canvas/canvas-context-menu.spec.browser2.tsx#L77